Shopify/cli
# Remove listr2 in favor of renderTasks everywhere

### WHY are these changes introduced?

Fixes https://github.com/Shopify/internal-cli-foundations/issues/401

We want to replace `listr2` with `renderTasks` everywhere.

### WHAT is this pull request doing?

This PR adds retry and skip logic to tasks handled by the `Tasks` component and replaces uses of listr2 in hydrogen commands.

TODO:
- [x] Add component tests for this new logic
- [x] Remove listr2 completely along with any code related to it (a lot of it inside `ui.ts`)

### How to test your changes?

- Run all the hydrogen commands
- See the new output

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
